crates/extensions/tedge_timer_ext/src/actor.rs

```diff
@@ -163,14 +164,13 @@ impl PartialEq for TimerEntry {
     }
 }
 
-impl From<TimerEntry> for (ClientId, Timeout<AnyPayload>) {
-    fn from(request: TimerEntry) -> Self {
-        (
-            request.client_id,
-            Timeout {
-                event: request.event_id,
-            },
-        )
+impl TimerEntry {
+    async fn trigger(self) -> Result<(), ChannelError> {
+        let response = Timeout {
+            event: self.event_id,
+        };
+        let mut sender = self.client_id;
+        sender.send(response).await
```

Comment:
> question: is the plan to remove this actor in the future?

It's not my plan. 

> If I understand correctly, this actor is used so that other actors can send a `TimeoutRequest` and this actor will send a message back after a specified time.

Your understanding is correct.

> Given that we want to use regular async/await without blocking, in order to reduce the amount of state we need to keep, for actors that spawn tasks for each request, they can trivially replace a call to this actor with just sleeping inline?

Spawning a task for each request is not always feasible. At least with the current design. As an example, once the `c8y_firmware_manager` has forwarded a request to a child-device, it has to wait for the child-device responses but not longer some timeout limit. What's prevent to handle this in the same task as for the download is that the message will be received over MQTT by the main actor and not the task. Somehow, the task has be promoted to
an actor (a receiver + a run method). Why not.

Reply:
Finally, I tried to implement this idea: https://github.com/thin-edge/thin-edge.io/pull/2600/commits/36ba09272717c40064010972f150424c5fc9efe0

The timer actor is no more used by the firmware actor and the whole firmware operation is implemented in a single async method (handling download, mqtt interaction with the child and timeout).